src/charm.py

```diff
@@ -128,10 +183,20 @@ def _relation_created(self, relation_name: str) -> bool:
             relation_name (str): Relation name
 
         Returns:
-            str: Whether the relation was created.
+            bool: Whether the relation was created.
         """
         return bool(self.model.get_relation(relation_name))
 
+    def _get_webui_endpoint_url(self) -> Optional[str]:
```

Comment:
```suggestion
    def _get_webui_endpoint_url(self) -> str:
```
This always return a string.

Reply:
✅  returning None now